pyfolio/perf_attrib.py

```diff
@@ -0,0 +1,138 @@
+            2017-01-01  0.002779 -0.005453
+            2017-01-02  0.001096  0.010290
+
+    factor_loadings : pd.DataFrame
+        Factor loadings for all days in the date range, with date and ticker as
+        index, and factors as columns.
+        - Example:
+                               momentum  reversal
+            dt         ticker
+            2017-01-01 AAPL   -1.592914  0.852830
+                       TLT     0.184864  0.895534
+                       XOM     0.993160  1.149353
+            2017-01-02 AAPL   -0.140009 -0.524952
+                       TLT    -1.066978  0.185435
+                       XOM    -1.798401  0.761549
+
+    Returns
+    -------
+    perf_attribution : pd.DataFrame
+        df with factors, common returns, and specific returns as columns,
+        and datetimes as index
+        - Example:
+                        momentum  reversal  common_returns  specific_returns
+            dt
+            2017-01-01  0.249087  0.935925        1.185012          1.185012
+            2017-01-02 -0.003194 -0.400786       -0.403980         -0.403980
+    """
+    # convert holdings to percentages, and convert positions to long format
+    positions = get_percent_alloc(positions)
+    # remove cash after normalizing positions
+    del positions['cash']
+    positions = positions.stack()
+    positions.index = positions.index.set_names(['dt', 'ticker'])
+
+    risk_exposures = factor_loadings.multiply(positions,
+                                              axis='rows')
+
+    risk_exposures_portfolio = risk_exposures.groupby(level='dt').sum()
+    perf_attrib_by_factor = risk_exposures_portfolio.multiply(factor_returns)
+
+    common_returns = perf_attrib_by_factor.sum(axis='columns')
+    specific_returns = returns - common_returns
+
+    returns_df = pd.DataFrame({'total_returns': returns,
+                               'common_returns': common_returns,
+                               'specific_returns': specific_returns})
+
+    return pd.concat([perf_attrib_by_factor, returns_df], axis='columns')
+
+
+def create_perf_attrib_stats(perf_attrib):
+    """
+    Takes perf attribution data over a period of time and computes annualized
+    multifactor alpha, multifactor sharpe, risk exposures.
+    """
+    summary = {}
+    specific_returns = perf_attrib['specific_returns']
+    common_returns = perf_attrib['common_returns']
+
+    summary['Annual multi-factor alpha'] =\
+        empyrical.annual_return(specific_returns)
+
+    summary['Multi-factor sharpe'] =\
+        empyrical.sharpe_ratio(specific_returns)
+
+    summary['Cumulative specific returns'] =\
+        empyrical.cum_returns(specific_returns)
+    summary['Cumulative common returns'] =\
+        empyrical.cum_returns(common_returns)
+    summary['Total returns'] =\
+        empyrical.cum_returns(perf_attrib['total_returns'])
+
+    summary = pd.Series(summary)
+    return summary
+
+
+def show_perf_attrib_stats(perf_attrib_data, risk_exposures):
```

Comment:
I think this would be more useful if it would take the same input arguments as `perf_attrib`, call `perf_attrib` and `create_perf_attrib_stats` and display the info. Perhaps alternatively it could work with pre-computed perf_attrib_data.